cf/mixin/fielddomainlist.py

```diff
@@ -0,0 +1,339 @@
+from ..functions import _DEPRECATION_ERROR_ARG
+
+
+class FieldDomainList:
+    """TODO."""
+
+    def select_by_construct(self, *identities, OR=False, **conditions):
+        """Select elements by their metadata constructs.
+
+        To find the inverse of the selection, use a list comprehension
+        with the !match_by_construct` method of the construct
+        elements. For example, to select all constructs that do *not*
+        have a "latitude" metadata construct:
+
+           >>> gl = cf.{{class}}(
+           ...     f for f in fl if not f.match_by_construct('latitude')
+           ... )
+
+        .. note:: The API changed at version 3.1.0
+
+        .. versionadded:: 3.0.0
+
+        .. seealso: `select`, `__call__`, `select_by_units`,
+                    `select_by_naxes`, `select_by_rank`,
+                    `select_by_property`
+
+        :Parameters:
+
+            identities: optional
+                Identify metadata constructs that have an identity,
+                defined by their `!identities` methods, that matches
+                any of the given values.
+
+                If no identities nor conditions (see the *conditions*
+                parameter) are provided then all constructs are
+                selected.
+
+                {{value match}}
+
+                {{displayed identity}}
+
+                If a cell method construct identity is given (such as
+                ``'method:mean'``) then it will only be compared with
+                the most recently applied cell method operation.
+
+                Alternatively, one or more cell method constructs may
+                be identified in a single string with a CF-netCDF cell
+                methods-like syntax for describing both the collapse
+                dimensions, the collapse method, and any cell method
+                construct qualifiers. If N cell methods are described
+                in this way then they will collectively identify the N
+                most recently applied cell method operations. For
+                example, ``'T: maximum within years T: mean over
+                years'`` will be compared with the most two most
+                recently applied cell method operations.
+
+                *Parameter example:*
+                  `'latitude'``
```

Comment:
This is a duplicate to the one a few down so can be removed(and the formatting needs fixing, so cutting this one rather than the latter is quickest).